src/ui/common/src/components/pages/workflow/id/index.tsx

```diff
@@ -221,8 +216,16 @@ const WorkflowPage: React.FC<WorkflowPageProps> = ({
         );
       }
 
-      for (const artfId of [...op.inputs, ...op.outputs]) {
```

Comment:
@vsreekanti @agiron123 This `...op.inputs` part confused me quite a bit. If we're looping through all operators, won't including both op's inputs _and_ outputs lead to a bunch of redundant artifact fetches??

Reply:
Yes, and I verified that after this fix, redundant fetches seem gone.